app/Models/Repositories/Product.js

```diff
@@ -0,0 +1,46 @@
+class Product {
+  static async attachAttrs(product, attributes) {
+    attributes.map(attribute =>
+      product.attributes().sync(attribute.id, row => {
+        row.value = attribute.value;
+      })
+    );
+
+    return Promise.all(attributes);
+  }
+
+  async attachAttrs(attributes) {
+    attributes.map(attribute =>
+      this.attributes().sync(attribute.id, row => {
```

Comment:
метод sync обновляет ID в таблице связи, такое поведение бывает или нейтральным или плохим, но никак не нормально. используй метод [attach](https://adonisjs.com/docs/4.1/relationships#_attach)

Reply:
Метод attach не обновляет существующие записи. Это делает только метод sync, т.к. он заточен похоже именно для этого, чтобы не писать детач и атач 

> The sync method is a shortcut for calling detach and then attach methods. It can be used for updating a pivot table.

Метод update тоже не подходит, т.к. он предназначен для relatedModel.
Поєтому оставляю так как есть, т.к. это работает, пусть даже этот способ и меняет primary_id в пивот таблице